### PR TITLE
Fix push issues: batch 1

### LINKED ISSUE #949: Created TEs are missing the `created_with` property in the JSON body
What the title says. The API rejects an `create` actions for TEs which don't contain `"created_with": "<some string>"`. The string should be the same as the user agent header.

### LINKED ISSUE #950: Fix processing user and preferences update action result
We're always processing the `user` and `preferences` action result even though it's `null` which throws an exception.

### PR DESCRIPTION
**Pull request by simonrozsival**
### Summary
This PR fixes two bugs I found this morning. I know about more issues with updating TEs which I haven't finished debugging yet and which will be addressed later.

### Relationships
Closes #949, closes #950 

### Pull Request checklist
- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [ ] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests

### Pointers
Try it yourself in the emulator!

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

## Merge permissions
🦑

### CODE REVIEW

**Review by heytherewill — APPROVED** [review]

**Review by heytherewill — CHANGES_REQUESTED** [review]
Actually, please add a test to prevent us from making the null user mistake again

File: sync/src/test/java/com/toggl/sync/processing/ResponseProcessorTest.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            clients = emptyList(),
+            tasks = emptyList(),
+            tags = emptyList(),
+            user = null,
+            preferences = null,
+        )
+
+        processor.process(response)
+
+        // if we reach the end of the test, no exception was thrown
```
Comment (daividssilverio) [inline sync/src/test/java/com/toggl/sync/processing/ResponseProcessorTest.kt:26]: There's this nice assertion that you could use here instead of leaving a comment
```
io.kotest.assertions.throwables.shouldNotThrowAny
```

File: sync/src/test/java/com/toggl/sync/processing/ResponseProcessorTest.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package com.toggl.sync.processing
+
+import com.toggl.api.network.models.sync.PushResponse
+import com.toggl.sync.BaseDatabaseTest
+import io.mockk.mockk
```
Comment (daividssilverio) [inline sync/src/test/java/com/toggl/sync/processing/ResponseProcessorTest.kt:5]:
```suggestion
import com.toggl.sync.BaseDatabaseTest
import io.kotest.assertions.throwables.shouldNotThrowAny
import io.mockk.mockk
```

File: sync/src/test/java/com/toggl/sync/processing/ResponseProcessorTest.kt (hunk 1 above)
Comment (daividssilverio) [inline sync/src/test/java/com/toggl/sync/processing/ResponseProcessorTest.kt:26]:
```suggestion
        shouldNotThrowAny {
            processor.process(response)
        }
```

**Review by heytherewill — APPROVED** [review]

**Review by daividssilverio — APPROVED** [review]
